### PR TITLE
Ability to generate planar metric terms and addition of `test_eta.py` unit test

### PR DESCRIPTION
**Pull request by fmalatino**
**Description**
A conditional statement has been added to the `__init__` method of the `MetricTerms` class to allow for generating surface level metric terms (no hybrid pressure values). Initialization of  such a `MetricTerms` object will now contain zero-valued hybrid pressures.

The unit test `test_eta.py` from `pace/tests/main/grid` has now been moved into `ndsl/tests/grid` along with a python module for generating the test eta level containing files.

This PR must be reviewed and merged after [PR 92](https://github.com/NOAA-GFDL/pace/pull/92) is merged in pace

**How Has This Been Tested?**
Tested via the tests contained within the `tests` directory of NDSL, and in the tests of the dependent repositories

Pace tests will fail as the changes from [PR 92](https://github.com/NOAA-GFDL/pace/pull/92) in Pace are necessary for passing.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included

### CODE REVIEW

**Comment by fmalatino** [comment]
Given that metric terms, and the grid generation methods as a whole are a part of NDSL, does it make sense to move the unit test `tests/main/grid/test_eta.py` in Pace to NDSL?

File: ndsl/grid/eta.py [hunk 1]
```diff
@@ -30,10 +30,8 @@ class HybridPressureCoefficients:
 
 
 def _load_ak_bk_from_file(eta_file: str) -> Tuple[np.ndarray, np.ndarray]:
-    if eta_file == "None":
-        raise ValueError("eta file not specified")
     if not os.path.isfile(eta_file):
-        raise ValueError("file " + eta_file + " does not exist")
+        raise ValueError("eta file does not exist")
```
Comment (FlorianDeconinck) [inline ndsl/grid/eta.py:34]: Keep `eta_file` in there, probably make a f-string to be more inline with the rest of the code

**Review by FlorianDeconinck — CHANGES_REQUESTED** [review]

**Comment by FlorianDeconinck** [comment]
> Given that metric terms, and the grid generation methods as a whole are a part of NDSL, does it make sense to move the unit test `tests/main/grid/test_eta.py` in Pace to NDSL?

Yes, please pull it down. We decided that the FV grid would come preset with NDSL, let's commit fully to it.

File: ndsl/grid/eta.py [hunk 2]
```diff
@@ -31,7 +31,7 @@ class HybridPressureCoefficients:
 
 def _load_ak_bk_from_file(eta_file: str) -> Tuple[np.ndarray, np.ndarray]:
     if not os.path.isfile(eta_file):
-        raise ValueError("eta file does not exist")
+        raise ValueError("eta file " + eta_file + " does not exist")
```
Comment (oelbert) [inline ndsl/grid/eta.py:34]:
```suggestion
        raise ValueError(f"eta file {eta_file} does not exist")
```

File: tests/grid/test_eta.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+def test_set_hybrid_pressure_coefficients_nofile():
+
+    """This test checks to see that the program
+    fails when (1) the eta_file is not specified in the yaml
+    configuration file; and (2), the computed eta values
+    increase non-monotonically.  For the latter test, the eta_file
+    is specified in test_config_not_mono.yaml file and
+    the ak and bk values in the eta_file have been changed nonsensically
+    to result in erronenous eta values.
```
Comment (oelbert) [inline tests/grid/test_eta.py:124]: It looks like this test only does the first of these two, so the docstring should reflect that

File: tests/grid/test_eta.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+def test_set_hybrid_pressure_coefficients_not_mono():
+
+    """This test checks to see that the program
+    fails when (1) the eta_file is not specified in the yaml
+    configuration file; and (2), the computed eta values
+    increase non-monotonically.  For the latter test, the eta_file
+    is specified in test_config_not_mono.yaml file and
+    the ak and bk values in the eta_file have been changed nonsensically
+    to result in erronenous eta values.
+    """
```
Comment (oelbert) [inline tests/grid/test_eta.py:177]: Same as above, please update the docstring to reflect the test

File: tests/grid/test_eta.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+    try:
+        metric_terms = MetricTerms(quantity_factory=quantity_factory, communicator=communicator, eta_file=eta_file)
+    except Exception as error:
+        if os.path.isfile(out_eta_file):
+            os.remove(out_eta_file)
+        if str(error) == "ETA values are not monotonically increasing":
+            pytest.xfail("testing eta values are not monotomincally increasing")
+        else:
+            pytest.fail(
+                "ERROR in testing etav values not are not monotonically increasing"
```
Comment (oelbert) [inline tests/grid/test_eta.py:224]:
```suggestion
                "ERROR in testing eta values not are not monotonically increasing"
```

**Review by oelbert — CHANGES_REQUESTED** [review]
Minor documentation changes mostly, otherwise lgtm

File: tests/grid/test_eta.py (hunk 3 above)
Comment (oelbert) [reply tests/grid/test_eta.py:124]: This docstring still needs to be updated

File: tests/grid/test_eta.py (hunk 4 above)
Comment (oelbert) [reply tests/grid/test_eta.py:177]: same with this one

File: ndsl/grid/generation.py [hunk 6]
```diff
@@ -297,12 +297,34 @@ def __init__(
         self._dy_center = None
         self._area = None
         self._area_c = None
-        (
-            self._ks,
-            self._ptop,
-            self._ak,
-            self._bk,
-        ) = self._set_hybrid_pressure_coefficients(eta_file, ak, bk)
+        if eta_file != "None":
```
Comment (oelbert) [inline ndsl/grid/generation.py:300]: Do we prefer `is not None` to `!- None`?

Comment (oelbert) [reply ndsl/grid/generation.py:300]: Also should we set the default vaule of eta_file to None instead of "None"?

File: ndsl/grid/generation.py [hunk 7]
```diff
@@ -297,12 +297,34 @@ def __init__(
...
+            (
+                self._ks,
+                self._ptop,
+                self._ak,
+                self._bk,
+            ) = self._set_hybrid_pressure_coefficients(eta_file, ak, bk)
+        else:
```
Comment (oelbert) [inline ndsl/grid/generation.py:307]: Is this the behavior we want? I thought we'd rather raise a ValueError or something here?

File: ndsl/grid/generation.py (hunk 6 above)
Comment (fmalatino) [reply ndsl/grid/generation.py:300]: That makes more sense I will push that shortly.

File: ndsl/grid/generation.py (hunk 7 above)
Comment (fmalatino) [reply ndsl/grid/generation.py:307]: A ValueError is raised deeper in when the file specified is either non-existent, and even lower if the values do not fit the size passed in the config. My thought was that we could still have hybrid pressure values; zero valued if no file passed, or specified when passed.

**Review by FlorianDeconinck — APPROVED** [review]
LGTM

File: ndsl/grid/generation.py [hunk 8]
```diff
@@ -237,7 +237,7 @@ def __init__(
         dy_const: float = 1000.0,
         deglat: float = 15.0,
         extdgrid: bool = False,
-        eta_file: str = "None",
+        eta_file: str = "",
```
Comment (FlorianDeconinck) [inline ndsl/grid/generation.py:240]: Default to `None`

File: ndsl/grid/generation.py [hunk 9]
```diff
@@ -297,12 +297,34 @@ def __init__(
         self._dy_center = None
         self._area = None
         self._area_c = None
-        (
-            self._ks,
-            self._ptop,
-            self._ak,
-            self._bk,
-        ) = self._set_hybrid_pressure_coefficients(eta_file, ak, bk)
+        if eta_file != "":
```
Comment (FlorianDeconinck) [inline ndsl/grid/generation.py:300]: Check `eta_file is not None`

**Review by FlorianDeconinck — CHANGES_REQUESTED** [review]
Cleanup the default eta_file using the pythonic `None`

File: ndsl/grid/generation.py (hunk 9 above)
Comment (fmalatino) [reply ndsl/grid/generation.py:300]: I had tried that initially, but the linting crashed.

File: ndsl/grid/generation.py (hunk 8 above)
Comment (fmalatino) [reply ndsl/grid/generation.py:240]: I had tried that initially, but the linting crashed.

File: ndsl/grid/generation.py (hunk 9 above)
Comment (fmalatino) [reply ndsl/grid/generation.py:300]: Pushed

File: ndsl/grid/generation.py [hunk 10]
```diff
@@ -237,7 +237,7 @@ def __init__(
         dy_const: float = 1000.0,
         deglat: float = 15.0,
         extdgrid: bool = False,
-        eta_file: str = "None",
+        eta_file: str = None,
```
Comment (FlorianDeconinck) [inline ndsl/grid/generation.py:240]: `Optional[str]`

**Review by FlorianDeconinck — CHANGES_REQUESTED** [review]
Fix a type hint and we are ready to go

File: ndsl/grid/generation.py (hunk 10 above)
Comment (fmalatino) [reply ndsl/grid/generation.py]: Pushed

**Review by FlorianDeconinck — APPROVED** [review]
LGTM

**Review by oelbert — APPROVED** [review]
👍

**Pull request merged**